src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -934,6 +1116,205 @@ - (void)readAttributesWithEndpointID:(NSNumber * _Nullable)endpointID
     std::move(*bridge).DispatchAction(self);
 }
 
+- (void)subscribeWithAttributePaths:(NSArray<MTRAttributeRequestPath *> * _Nullable)attributePaths
+                         EventPaths:(NSArray<MTREventRequestPath *> * _Nullable)eventPaths
+                             params:(MTRSubscribeParams * _Nullable)params
+                              queue:(dispatch_queue_t)queue
+                      reportHandler:(MTRDeviceResponseHandler)reportHandler
+            subscriptionEstablished:(MTRSubscriptionEstablishedHandler _Nullable)subscriptionEstablished
+{
+    if ((attributePaths == nil || [attributePaths count] == 0) && (eventPaths == nil || [eventPaths count] == 0)) {
+        dispatch_async(queue, ^{
+            reportHandler(nil, [MTRError errorForCHIPErrorCode:CHIP_ERROR_INVALID_ARGUMENT]);
+        });
+        return;
+    }
+
+    if (self.isPASEDevice) {
+        // We don't support subscriptions over PASE.
+        dispatch_async(queue, ^{
+            reportHandler(nil, [MTRError errorForCHIPErrorCode:CHIP_ERROR_INCORRECT_STATE]);
+        });
+        return;
+    }
+
+    // Copy params before going async.
+    attributePaths = (attributePaths == nil) ? nil : [attributePaths copy];
+    eventPaths = (eventPaths == nil) ? nil : [eventPaths copy];
```

Comment:
This is not in fact copying them usefully; it's just grabbing references to the same objects in a new array.  This should probably actually create new (mutable) arrays with the right capacity and then fill them with copies.

Reply:
I modified.